### PR TITLE
Config documentation generation fixes for #5144 and #5703

### LINKED ISSUE #5144: generate a documentation file for each config root and general config items (non config group config items)
**Description**
Write now we generate:
- an extension documentation file (extension config root combined)
- a config group configuration file (each config group)
- all configuration file (all configuration roots)

It could be nice to also generate a file per config root and general config items (non config group config items)

**Implementation ideas**
- the name of the file for config root could be `quarkus-<config-root-name>-<classNameWithout deployment/runtime>.adoc` e.g `quarkus-datasource-agroal-build-time-config.adoc`, `quarkus-datasource-agroal-runtime-config.adoc`
- the name of the file for general config items  could be `quarkus-<extension-name>-general-config-items.adoc` e.g `quarkus-agroal-general-config-items-adoc`, `quarkus-hibernate-orm-general-config-items.adoc`

/cc @gsmet

### LINKED ISSUE #5703: Consider Optional Config Group and Sets in config doc generation
https://github.com/quarkusio/quarkus/pull/5387 comes with so many goodness like added support for Optional config group and Sets. The config doc generator must also improve in the same sense and must recognise these two constructs.

_Quick ideas._ 
- mark all config items inside a optional group as as optional
- do not display type information for Set but `list of` or `set of` ?

/cc @dmlloyd as discussed here https://github.com/quarkusio/quarkus/pull/5387#issuecomment-557133752

### PR DESCRIPTION
**Pull request by machi1990**
Supersedes https://github.com/quarkusio/quarkus/pull/5190

Fixes #5144 

Fixes #5703

Follows up #5044 and #5387

/cc @gsmet @dmlloyd @FroMage

### CODE REVIEW

File: core/processor/src/main/java/io/quarkus/annotation/processor/generate_doc/ConfigDoItemFinder.java [hunk 1]
```diff
@@ -162,19 +163,19 @@ ScannedConfigDocsItemHolder findInMemoryConfigurationItems() {
 
             if (isConfigGroup) {
                 List<ConfigDocItem> groupConfigItems = recordConfigItemsFromConfigGroup(configPhase, name, configGroup,
-                        configSection, withinAMap, sectionLevel);
+                        configSection, withinAMap, sectionLevel, optional);
```
Comment (dmlloyd) [inline core/processor/src/main/java/io/quarkus/annotation/processor/generate_doc/ConfigDoItemFinder.java]: I don't think it is correct to make all members of an optional group be optional.  In fact an optional config group can have required members.  In this case, the required members are all required if any member of the group was given.

**Review by dmlloyd — CHANGES_REQUESTED** [review]
I found one issue, the rest seems okay though.

File: core/processor/src/main/java/io/quarkus/annotation/processor/generate_doc/ConfigDoItemFinder.java (hunk 1 above)
Comment (machi1990) [reply core/processor/src/main/java/io/quarkus/annotation/processor/generate_doc/ConfigDoItemFinder.java]: That's true, thanks for clarifying the behaviours. 

We can change this to maybe add a section indicating that it is optional. The members will be left untouched. WDYT?

Comment (dmlloyd) [reply core/processor/src/main/java/io/quarkus/annotation/processor/generate_doc/ConfigDoItemFinder.java]: Sounds good.

File: docs/src/main/asciidoc/building-native-image.adoc [hunk 2]
```diff
@@ -398,7 +398,7 @@ These are provided in `application.properties` the same as any other config prop
 
 The properties are shown below:
 
-include::{generated-dir}/config/quarkus-core-native-config.adoc[opts=optional]
+include::{generated-dir}/config/quarkus-native-pkg-native-config.adoc[opts=optional]
```
Comment (gsmet) [inline docs/src/main/asciidoc/building-native-image.adoc]: What's the rationale of this change?

Comment (machi1990) [reply docs/src/main/asciidoc/building-native-image.adoc]: To use a generated `NativeConfig` root file. The file name is not handled derived from the package name by stripping out `runtime` or `deployment`.

**Review by FroMage — APPROVED** [review]
LGTM

**Review by dmlloyd — APPROVED** [review]

**Pull request merged**